pkg/platform/kube/test/platform_test.go

```diff
@@ -1398,6 +1398,26 @@ def handler(context, event):
 	suite.DeployFunctionAndRedeploy(createFunctionOptions, afterFirstDeploy, afterSecondDeploy)
 }
 
+func (suite *DeleteFunctionTestSuite) TestProcessorShutdown() {
```

Comment:
This is a nice test that verifies the processor is doing its expected flow signaling the triggers, but I would also want to see that the draining is actually executed in the runtimes, to make sure that the timeout is waited and we do let the user-code finish its work.

I would suggest creating a test similar to `TestTerminationHook` in kafka, utilizing the existing `termination-hook.py` example function, where you verify that when you delete the pod the runtimes does what is expected.

Reply:
Thanks, make sense!
It was decided to add such a test as a separate PR so as not to mix everything together.